### PR TITLE
feat: added support for embeddings via HuggingFace Inference API

### PR DESCRIPTION
**Pull request by tasmay**
Added support for generating embeddings via the HuggingFace Inference API. The same exists in the Python version but missing in LlamaIndexTS.

### CODE REVIEW

File: packages/core/src/embeddings/HuggingFaceEmbedding.ts [hunk 1]
```diff
@@ -46,3 +47,64 @@ export class HuggingFaceEmbedding extends BaseEmbedding {
     return Array.from(output.data);
   }
 }
+
+// TODO workaround issue with @huggingface/inference@2.7.0
```
Comment (marcusschiesser) [inline packages/core/src/embeddings/HuggingFaceEmbedding.ts:51]: maybe move this to `HuggingFaceInferenceAPIEmbedding`? as it's very different to the inference via transformers.js?

Comment (marcusschiesser) [inline packages/core/src/embeddings/HuggingFaceEmbedding.ts:51]: is there still a TODO?

Comment (tasmay) [reply packages/core/src/embeddings/HuggingFaceEmbedding.ts:51]: @huggingface/inference no longer exports Options, if they fix it in later versions, we can switch to using it.

Comment (himself65) [reply packages/core/src/embeddings/HuggingFaceEmbedding.ts:51]: link to an issue?

Comment (tasmay) [reply packages/core/src/embeddings/HuggingFaceEmbedding.ts:51]: I am not sure if there is an open issue regarding this. I kept the implementation similar to [llm/huggingface.ts](https://github.com/run-llama/LlamaIndexTS/blob/01658fdb31efdf705519b6072f689182c3fdc87f/packages/llamaindex/src/llm/huggingface.ts#L20) @huggingface/inference has not exported Options in their latest version. If I do:

```
import { type Options as HfInferenceOptions } from "@huggingface/inference"
```
gives an error: Module '"@huggingface/inference"' has no exported member 'Options'. Hence, this workaround.

Comment (himself65) [reply packages/core/src/embeddings/HuggingFaceEmbedding.ts:51]: Oh I see, you can try this to infer type from it
 
```ts
type Option = ConstructorParameters<...>[1]
```

**Review by marcusschiesser — APPROVED** [review]
Thanks @tasmay

**Comment by marcusschiesser** [comment]
@tasmay looks good, please resolve conflicts, then we can merge it

**Comment by tasmay** [comment]
Done. Thanks @marcusschiesser and @himself65

**Review by himself65 — APPROVED** [review]

**Pull request merged**